### PR TITLE
(SIMP-1695) Add set of Port Data Types

### PR DESCRIPTION
**Pull request by trevor-vaughan**
This adds the following Data Types:
  * Port (any of the below)
  * RandomPort (0)
  * PrivilegedPort (1-1024)
  * UnprivilegedPort (1025-65535)

SIMP-1695 #comment Required for Puppet 4 refactor

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,9 @@
 * Wed Dec 07 2016 Trevor Vaughan <tvaughan@onyxpoint.com> - 2.1.0-0
+- Added a set of Data Types
+  - Port (any of the below)
+  - RandomPort (0)
+  - PrivilegedPort (1-1024)
+  - UnprivilegedPort (1025-65535)
```
Comment (lamawithonel) [inline CHANGELOG]: According to [RFC 6335, Section 6](https://tools.ietf.org/html/rfc6335#section-6), `PrivilegedPort` should be `SystemPort` and `UnprivilegedPort` should be `UserPort`. Also, User Ports only go up to 49151; from 49152 to 65535 are "Dynamic Ports." You could probably leave out the dynamic ports for now, if time is an issue.

Comment (trevor-vaughan) [inline CHANGELOG]: @lamawithonel Challenge accepted...mind submitting a PR?

Comment (lamawithonel) [inline CHANGELOG]: PR submitted against your PR branch.

**Comment by lamawithonel** [comment]
Hmmm... spec helpers doesn't run the tests.

**Pull request merged**